openqa-qam-sle

```diff
@@ -0,0 +1,208 @@
+            _warn "RUNNING / SCHEDULED ($running jobs) Awaiting completion..."
+            echo ""
+            continue
+        }
+
+        local failed_url="${url_openqa}/api/v1/jobs/overview?distri=sle&version=${version}&build=${build}&groupid=${groupid}&result=failed&result=incomplete&result=timeout_exceeded"
+        local failed
+        failed=$(curl -sk -X GET "$failed_url" | jq -r '. | length')
+        [[ $failed -eq 0 ]] && _ok "PASSED"
+        [[ $failed -gt 0 ]] && _ko "FAILED ($failed jobs)"
+        echo ""
+    done
+    echo "---"
+}
+
+search_maintenance_aggregated() {
+    _title "Maintenance: Aggregated updates / Core Maintenance Updates"
+    local review_request_id=$1
+    local days=${2:-5}  # Default last 5 days
+    local groupid="414" # Maintenance: Aggregated updates / Core Maintenance Updates
+
+    local ID
+    ID="$(echo "$review_request_id" | awk -F':' '{printf "%s", $3}')" # The ID of the update
+
+    local versions
+    versions=$(curl -sk -X GET "${url_dashboard_qam}/api/incident_settings/$ID" | jq -r '.[].settings.VERSION' | sort | uniq)
+
+    for version in $versions; do
+        [[ -z "${dict_group[$version]}" ]] && continue
+
+        for day in $(seq 0 "$days"); do
+            local build
+            build="$(date +%Y%m%d -d "$day day ago")-1"
+
+            local URL="${url_openqa}/api/v1/jobs/overview?distri=sle&version=${version}&build=${build}&groupid=${groupid}&state=done"
+
+            local id
+            id=$(curl -sk -X GET "$URL" | jq -r '.[0].id | select( . != null)')
+
+            [[ -z "$id" ]] && continue
+
+            local json
+            json=$(curl -sk -X GET "${url_openqa}/api/v1/jobs/${id}")
+
+            local issues
+            issues=$(echo "$json" | jq -r '.job.settings | to_entries | .[] | .key + "=" + (.value | @sh)' | grep -E '_TEST_ISSUES=' | perl -lpe "s/.*=//;s/,/ /g" | tr -d "'")
+
+            for issue in $issues; do
+                # echo "- Issue: $issue"
+                [[ $issue -eq $ID ]] && {
+                    echo "Version: '$version' Update: '$ID'"
+                    echo "Build $build contains $ID"
+                    echo "${url_openqa}/tests/overview?distri=sle&version=${version}&build=${build}&groupid=${groupid}"
+
+                    local running_url="${url_openqa}/api/v1/jobs/overview?distri=sle&version=${version}&build=${build}&groupid=${groupid}&state=scheduled&state=running"
+                    local running
+                    running=$(curl -sk -X GET "$running_url" | jq -r '. | length')
+                    [[ $running -gt 0 ]] && {
+                        _warn "RUNNING / SCHEDULED ($running jobs) Awaiting completion..."
+                        echo ""
+                        continue
+                    }
+
+                    local failed_url="${url_openqa}/api/v1/jobs/overview?distri=sle&version=${version}&build=${build}&groupid=${groupid}&result=failed&result=incomplete&result=timeout_exceeded"
```

Comment:
I suggest you call that "not_ok" instead of "failed" as you include more than just "failed"

Reply:
I know. To make that not confusing I was suggesting to use a different type